services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/HankeServiceImpl.kt

```diff
@@ -899,4 +918,39 @@ open class HankeServiceImpl(
         )
         loggingEntryHolder.saveLogEntries(auditLogService)
     }
+
+    private fun Application.withHanketunnus(hankeTunnus: String) =
+        Application(
+            id,
+            alluid,
+            alluStatus,
+            applicationIdentifier,
+            applicationType,
+            applicationData,
+            hankeTunnus
+        )
+
+    /** Autogenerated hanke based on application data. Generated flag true. */
+    private fun generateHankeFrom(application: Application): Hanke =
+        createHanke(
+            Hanke(
+                null,
+                null,
+                null,
+                application.applicationData.name,
+                null,
+                null,
+                null,
+                null,
+                null,
+                null,
+                null,
+                null,
+                null,
+                null,
+                HankeStatus.DRAFT,
+                null,
+                true,
```

Comment:
Adding the parameter names to this call makes it a lot easier to see e.g. which parameter is set to true. Idea can even do it for you, if you ask nicely.